dask/dataframe/core.py

```diff
@@ -3420,6 +3420,19 @@ def map(self, arg, na_action=None, meta=no_default, is_monotonic=False):
             applied = applied.clear_divisions()
         return applied
 
+    def _get_names(self):
+        return self._meta.names
+
+    @derived_from(pd.Index)
+    def set_names(self, names, level=None, inplace=False):
+        if inplace:
+            raise NotImplementedError("The inplace= keyword is not supported")
+        meta = self._meta_nonempty.set_names(names, level)
+        result = self.map_partitions(M.set_names, names, level, inplace, meta=meta)
+        return result
+
+    names = property(fset=set_names, fget=_get_names)
```

Comment:
Hmm I don't think this is what we want. IIUC, doing

```python
df.index.names = ['a', 'b']
```

would operate on a new index object, rather than updating `df.index`. Is that correct?

Reply:
You are right, it is not covering the user case. I noticed the df.index is a property that always returns a new index, that is why the PR does not support any inplace operation.
I have a new commit that caches the df.index and returns it if it exists, otherwise, it does exactly the same as before.
This way it is possible to have inplace operations, but I am not sure if it will cover all the cases. I'll appreciate any feedback you have. Thanks!